lib/apollo-federation/schema.rb

```diff
@@ -12,49 +12,74 @@ def self.included(klass)
     end
 
     module ClassMethods
-      def to_graphql
-        orig_defn = super
-
-        possible_entities = orig_defn.types.values.select do |type|
-          !type.introspection? && !type.default_scalar? && type.is_a?(GraphQL::ObjectType) &&
-            type.metadata[:federation_directives]&.any? { |directive| directive[:name] == 'key' }
-        end
+      def query(new_query_object = nil)
+        if new_query_object
+          @orig_query_object = new_query_object
+        else
+          if !@federation_query_object
+            # Build the new query object with the '_service' field
+            @federation_query_object = federation_query(@orig_query_object)
 
-        @query_object = federation_query
+            # If there are any "entities", define the Entity union and and the Query._entities field
+            possible_entities = schema_entities
+            if !possible_entities.empty?
+              entity_type = Class.new(Entity) do
+                possible_types(*possible_entities)
+              end
+              # TODO: Should/can we encapsulate all of this inside the module? What's the best/most
+              # Ruby way to split this out?
+              @federation_query_object.define_entities_field(entity_type)
```

Comment:
Defining the `_entities` field like this sets the correct parent to the type. This is not true with the `_service` field. Can we refactor how the `_service` field is included an do the same as `_entities`.

Basically, in this method we would call at some point:

```
@federation_query_object.define_service_field
```

Reply:
Hmmm. Can you clarify what you mean by "parent"? Is the missing parent causing issues? Also, (like above) is this something that was already broken in v1.0.4?

Reply:
Sorry, I should have said `owner`, not `parent`. If I recall correctly this `self` is evaluated as the module and not the query type.

https://github.com/rmosolgo/graphql-ruby/blob/master/lib/graphql/schema/member/has_fields.rb#L12